tests/TestCase/Database/Type/DateTimeTypeTest.php

```diff
@@ -308,6 +308,26 @@ public function testMarshal($value, $expected)
         }
     }
 
+    public function testMarshalWithLocalTimezone()
+    {
+        $this->type->setLocalTimezone('+0200');
+
+        $value = '2020-05-01 23:28:00';
+        $expected = new Time($value);
+
+        $result = $this->type->marshal($value);
+        $this->assertEquals($expected, $result->addHours(2));
+
+        $expected = new Time('2020-05-01 23:28:00', '+0200');
```

Comment:
Shouldn't the output time of marshalling be in application's timezone?

Reply:
Yup, I had forgotten about datetime array initially since it's no longer used in Cake 4. I'll update the related code and test the case.